core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -388,129 +379,87 @@ private DefaultFunctionResolver addtime() {
             nullMissingHandlingWithProperties(DateTimeFunction::exprAddTime), TIME, TIME, TIME),
         implWithProperties(
             nullMissingHandlingWithProperties(DateTimeFunction::exprAddTime), TIME, TIME, DATE),
-        implWithProperties(
-            nullMissingHandlingWithProperties(DateTimeFunction::exprAddTime), TIME, TIME, DATETIME),
         implWithProperties(
             nullMissingHandlingWithProperties(DateTimeFunction::exprAddTime),
             TIME,
             TIME,
             TIMESTAMP),
         implWithProperties(
             nullMissingHandlingWithProperties(DateTimeFunction::exprAddTime),
-            DATETIME,
-            DATETIME,
+            TIMESTAMP,
+            DATE,
```

Comment:
Should this have not been replaced by `TIMESTAMP, TIMESTAMP`?

Reply:
The Github changes on this is a little bit weird but I am basically just removing everything with a DATETIME it it. Before there were 16 combinations ( (T, T/D/DT/TS) + (D/DT/TS, T/D/DT/TS) ), and now there are just 9 combinations ( (T, T/D/TS) + (D/TS, T/D/TS) ) which my updated code covers.

Reply:
I see! Thanks for the explanation!